{{cookiecutter.project_name}}/steps/predict.py

```diff
@@ -1,7 +1,32 @@
 import mlflow
 from pyspark.sql.functions import struct, lit, to_timestamp
 
+{% if cookiecutter.include_feature_store == "yes" %}
+def predict_batch(
+    fs_client, spark_session, model_uri, input_table_name, output_table_name, model_version, ts
+):
+    """
+    Apply the model at the specified URI for batch inference on the table with name input_table_name,
+    writing results to the table with name output_table_name
+    """
+    table = spark_session.table(input_table_name)
 
+    prediction_df = fs_client.score_batch(
+        model_uri,
+        table
+    )
```

Comment:
With or without feature store, the only difference is the predict step.
You can keep BatchInference file the same and only modify inside the predict_batch function.

```
...
table = spark_session.table(input_table_name)

{% if cookiecutter.include_feature_store == "yes" %}
    from databricks.feature_store import FeatureStoreClient

    fs_client = FeatureStoreClient()
    prediction_df = fs_client.score_batch(
        model_uri,
        table
    )
    prediction = prediction_df["prediction"]
{% else %}
    predict = mlflow.pyfunc.spark_udf(
        spark_session, model_uri, result_type="string", env_manager="conda"
    )
    prediction = predict(struct(*table.columns))
{% endif %}

    output_df = (
        table.withColumn("prediction", prediction)
    ...
```

Reply:
sounds good

Reply:
Aren't all the steps cleaned up if feature store is enabled? Can you please double check?

Reply:
@vladimirk-db Great question! We are leaving the steps in the repo when feature store is enabled. Partial reason is that the folder also contains functions not related to recipes, for example, deploy.py, utils.py.
After the refactor-layout PR, the steps folder will only contain recipes functions and we can remove it.

Reply:
Sounds good, just making sure we don't prune this file by accident :-)